akka-http-core/src/main/scala/akka/http/impl/util/JavaMapping.scala

```diff
@@ -180,6 +180,9 @@ private[http] object JavaMapping {
   class Inherited[J <: AnyRef, S <: J](implicit classTag: ClassTag[S]) extends JavaMapping[J, S] {
     def toJava(scalaObject: S): J = scalaObject
     def toScala(javaObject: J): S = cast[S](javaObject)
+
+    import scala.language.higherKinds
+    def upcast[F[+_]](f: F[J]): F[S] = f.asInstanceOf[F[S]]
```

Comment:
or should it be `downcast`?

Reply:
I guess it's a (safe) downcast https://en.wikipedia.org/wiki/Downcasting

Reply:
Woops, thanks, fixed.